ami/data/buffers/random_data_buffer.py

```diff
@@ -0,0 +1,76 @@
+from typing import Any, Self
+
+import numpy as np
+import torch
+from torch.utils.data import TensorDataset
+
+from ..step_data import DataKeys, StepData
+from .base_data_buffer import BaseDataBuffer
+
+
+class RandomDataBuffer(BaseDataBuffer):
+    """A data buffer which does not preserve data order."""
+
+    def init(self, max_len: int, key_list: list[DataKeys]) -> None:
+        """Initializes data buffer.
+
+        Args:
+            max_len (int): max length of buffer.
+            key_list (list[DataKeys]): a list of keys to save whose values to buffer.
+        """
+        self.__max_len = max_len
+        self.__current_len = 0
+        self.__key_list = key_list
+        self.__buffer_dict: dict[DataKeys, list[Any]] = dict()
+        for key in key_list:
+            self.__buffer_dict[key] = []
+
+    def __len__(self) -> int:
+        """Returns current data length.
+
+        Returns:
+            int: current data length.
+        """
+        return self.__current_len
+
+    def add(self, step_data: StepData) -> None:
+        """Add a single step of data.
+
+        Args:
+            step_data (StepData): A single step of data.
+        """
+        if self.__current_len < self.__max_len:
+            for key in self.__key_list:
+                self.__buffer_dict[key].append(step_data[key])
```

Comment:
```suggestion
                v = step_data[key]
                assert isinstance(v, Tensor), "Got data must be Tensor."
                self.__buffer_dict[key].append(v)
```
値がAny型なのでTensorかどうかをチェックする必要があります。`torch.stack`で結局エラーが出るからですね。ただmake datasetが呼ばれるのはしばらくランタイムが経過してからと想定されるので、今回の修正のような早期のエラー判定が必要になります。

Reply:
`torch.tensor`によって任意のデータをテンソル型へ変換するようにしました。
0dec79b: Dataをappendする際にTensorに変換